### ISSUE TITLE
BEM conventions for all name formats

### ISSUE THREAD

**Issue by ncoden** [sasstools/sass-lint#598]
Hi

I use in my projects a BEM convention for both classes, variables, mixins and placeholders. I think it could be usefull to allow `strictbem` and `hyphenatedbem` conventions on them.

For exemple, a ~~*stupid*~~ use case:
```scss
$my-block__color--blue: blue;
$my-block__color--red: red;

@mixin my-block--blue {
  ...
}
@mixin my-block--red {
  ...
}

.my-block {
  ...

  // default
  @include my-block--blue;

  // modifiers
  &--blue { @include my-block--blue; }
  &--red  { @include my-block--red; }
}
```

Nicolas

**Comment by DanPurdy** [comment]
@benthemonkey what do you think? I know you handled these rules previously.

**Comment by ncoden** [comment]
thank you for the quick response :)
On other repos much more popular, I have to wait for days ! (yes, i'm thinking of you Foundation)

**Comment by benthemonkey** [comment]
@ncoden See our [docs](https://github.com/sasstools/sass-lint/blob/develop/docs/rules/class-name-format.md), we support `strictbem` and `hyphenatedbem` according to the same RegEx's as scss-lint.

Sorry for the delay.

**Comment by DanPurdy** [comment]
@benthemonkey we dont for variables, mixins or placeholders though, I think that's the issue here.

**Comment by benthemonkey** [comment]
Ah, sorry. Yeah I think it makes sense for all the format related rules to have identical configs.

**Comment by ncoden** [comment]
Ok. Can I make a PR or do you take care of that @benthemonkey ?

**Comment by benthemonkey** [comment]
@ncoden go for it. Please be sure to update the docs as well. Thanks!

**Comment by ncoden** [comment]
Pull request: https://github.com/sasstools/sass-lint/pull/604
poke @benthemonkey

**Comment by DanPurdy** [comment]
This is available and resolved in sass-lint 1.6.0
